pkg/kubectl/resource/builder.go

```diff
@@ -584,6 +584,9 @@ func (b *Builder) visitorResult() *Result {
 			if b.flatten {
 				visitors = NewFlattenListVisitor(visitors, b.mapper)
 			}
+			if b.defaultNamespace {
+				visitors = NewDecoratedVisitor(visitors, SetNamespace(b.namespace))
+			}
```

Comment:
I don't see why these code are related to this PR. Could you shed some light? Thank you.